libraries/thermocouple/Thermocouple.cpp

```diff
@@ -3,26 +3,37 @@
   Created by Vamshi Balanaga, Sept 18, 2020.
 */
 
-#include "therm.h"
+#include "Thermocouple.h"
 
 Thermocouple::Thermocouple(int numSensors){
   _ow = OneWire(OW_DATA_PIN);
   numThermos = numSensors;
-  _ow.reset();
+
+  // for (int i = 0; i < 8; i++){
+  //   Serial.print(rom[i]);
+  //   Serial.print(" ");
+  // }
+  // Serial.println("");
+
 }
 
+void Thermocouple::init(){
+  _ow.reset();
+  delay(250);
+  _ow.search(rom);
+}
 /**
  * whichSensor is 0 indexed.
  */
 void Thermocouple::readTemperatureData(float *data, int whichSensor) {
-  if(whichSensor >= numSensors){
+  if(whichSensor >= numThermos) {
     data[0] = -1;
     return;
   }
 
   scrollToRightSensor(whichSensor);
 
-  for (int i = 0; i < 9; i++){ // need 9 bytes apparently.
+  for (int i = 0; i < 9; i++) { // need 9 bytes apparently.
```

Comment:
I see it's reading 9 bytes but each float is only 4 bytes in the way I've implemented the I2C communication. Just wanted to make sure this won't be a problem.

Reply:
That's fine. See the lines below, not all of that data is the temperature reading.
Some of it is other metadata